lib/section.zsh

```diff
@@ -46,9 +46,18 @@ _compose_prompt() {
   # http://zsh.sourceforge.net/Doc/Release/Conditional-Expressions.html
   setopt EXTENDED_GLOB LOCAL_OPTIONS
 
-  # Threat the first argument as list of prompt sections
+  # Treat the first argument as list of prompt sections
   # Compose whole prompt from diferent parts
+  # If section is a function then invoke it
+  # If the composed name spaceship_$section is a function invoke it
   for section in $@; do
-    spaceship_$section
+    if typeset -f + $section &> /dev/null; then
```

Comment:
I think `typeset -f + $section` succeeds if `$section` is empty, which is not what we want. How about we just use `which` to test if it's executable? I don't see any reason to disallow out-of-process plugins (although I might discourage them for performance reasons).

Reply:
It is correct, but I don't think it will happen in this case because were iterating over arguments coming from an array. Empty strings are omitted I believe (not sure). I will enclose `$section` in a string.

Reply:
@bric3 have you considered the case when section equals `node`, for example. We check if `node` is available to invoke and invoke `node` binary, despite the fact that we just want to execute `spaceship_node`. Would it work that way or `typeset -f + $section` is smart enough to check only _declared functions_?

Reply:
I'm not sure what you mean by `node`. However `typeset -f` only reports defined function. I will push later an update to this code regarding @nfischer's comment.

Reply:
>  I will enclose $section in a string.

This fixes it in my zsh, thanks!

> However typeset -f only reports defined function

This is probably a wise idea, it should address @denysdovhan's concern.

I still think a good extra-precaution is to require the `spaceship_` prefix, since it reduces the risk that someone will accidentally invoke a command that's not intended for prompts. But this would require wrapper functions (which @bric3 seems to dislike). For my purposes, either approach is OK.

Reply:
Yes I think software should make life easier. If code can make user life easier that is a win. I don't think _wrapper functions_ are elegant in this situation. However I'm sold to some variable to declare used external functions.